File: client/java/src/avt/CellFarm.java
package avt;

import javax.microedition.lcdui.Graphics;
import main.Canvas;

public final class CellFarm extends SubObject {
   public int xCell;
   public int yCell;
   public int idTree;
   public short time;
   public short time1 = -1;
   public long tempTime;
   public boolean isArid;
   public boolean isWorm;
   public boolean isGrass;
   public boolean isSelected = false;
   public byte hervestPer;
   public byte vitalityPer;
   public byte status;
   public byte level;
   public int statusTree;

   public final void paint(Graphics var1) {
      if (super.x * MyObject.hd >= AvCamera.gI().xCam - 10 && this.xCell * MyObject.hd <= AvCamera.gI().xCam + Canvas.w + 10) {
         if (this.isSelected) {
            var1.drawImage(FarmScr.imgSell, (super.x - 13) * MyObject.hd, (super.y - 18) * MyObject.hd, 0);
         }

         int var2 = FarmScr.focusCell.x;
         int var3 = FarmScr.focusCell.y;
         if (this.xCell == var2 && this.yCell == var3) {
            Canvas.smallFontYellow.drawString(var1, "lv" + this.level, super.x * MyObject.hd, (super.y - 44) * MyObject.hd, 2);
         }

         if (this.idTree != -1) {
            TreeInfo var4;
            (var4 = FarmData.getTreeByID(this.idTree)).paint(var1, this.statusTree, super.x * MyObject.hd, super.y * MyObject.hd, 33);
            int var5 = var4.harvestTime * 60 + var4.dieTime * 60;
            if ((this.time <= var5 || var4.dieTime == -1) && this.hervestPer != 100 && this.time >= 0) {
               if (this.isGrass) {
                  var1.drawImage(FarmScr.imgWorm_G[1], (super.x + 5) * MyObject.hd, (super.y - 12) * MyObject.hd, 3);
               }

               if (this.isWorm) {
                  var1.drawImage(FarmScr.imgWorm_G[0], (super.x - 7) * MyObject.hd, super.y * MyObject.hd, 3);
               }

               if (this.xCell == var2 && this.yCell == var3 || this.time1 != -1 && this.time1 == FarmScr.idSelected) {
                  var2 = var2 * 24 * MyObject.hd;
                  var3 = var3 * 24 * MyObject.hd;
                  if (this.time1 != -1 && this.time1 == FarmScr.idSelected) {
                     var2 = this.xCell * 24 * MyObject.hd;
                     var3 = this.yCell * 24 * MyObject.hd;
                  }

                  var4.paint(var1, 7, var2 - 3, var3 - 40 * MyObject.hd, 33);
                  var1.setColor(1);
                  var1.fillRect(var2 - 4 * MyObject.hd, var3 - 38 * MyObject.hd, 31 * MyObject.hd, 5 * MyObject.hd);
                  var1.setColor(65280);
                  var1.fillRect(var2 - 3 * MyObject.hd, var3 - 37 * MyObject.hd, this.vitalityPer * 30 / 100 * MyObject.hd, 3 * MyObject.hd);
                  var1.setColor(2512938);
                  var1.drawRect(var2 - 4 * MyObject.hd, var3 - 38 * MyObject.hd, 31 * MyObject.hd, 4 * MyObject.hd);
                  long var10 = (long)(var4.harvestTime * 60 - this.time);
                  long var8 = var10 * 60L;
                  String var20 = "";
                  if (var8 < 0L) {
                     var8 = 0L;
                  }

                  long var13 = var8 / 60L / 60L;
                  long var15 = var8 / 60L % 60L;
                  long var17 = var8 % 60L;
                  var20 = var20 + var13 + ":" + var15 + ":" + var17;
                  if (var10 <= 0L) {
                     var20 = T.canHarvest;
                  }

                  Canvas.smallFontYellow.drawString(var1, var20, var2 + 5 * MyObject.hd, var3 - 49 * MyObject.hd, 0);
                  if ((var5 = this.time * 100 / (var4.harvestTime * 60) * 30 / 100) == 0) {
                     var5 = 1;
                  }

                  if (var5 >= 30) {
                     var5 = 29;
                  }

                  if (var4.harvestTime * 60 - this.time < 0) {
                     var5 = 30;
                  }

                  var1.setColor(1);
                  var1.fillRect(var2 - 4 * MyObject.hd, var3 - 32 * MyObject.hd, 31 * MyObject.hd, 5 * MyObject.hd);
                  var1.setColor(255, 255, 0);
                  var1.fillRect(var2 - 3 * MyObject.hd, var3 - 31 * MyObject.hd, var5 * MyObject.hd, 3 * MyObject.hd);
                  var1.setColor(2512938);
                  var1.drawRect(var2 - 4 * MyObject.hd, var3 - 32 * MyObject.hd, 31 * MyObject.hd, 4 * MyObject.hd);
                  byte var19 = 0;
                  if (this.isGrass) {
                     var19 = 1;
                     FarmScr.q.drawFrame(1, var2 + (5 + (this.isWorm ? 6 : 0)) * MyObject.hd, var3 - 22 * MyObject.hd, 0, var1);
                  }

                  if (this.isWorm) {
                     FarmScr.q.drawFrame(0, var2 + (4 - var19 * 6) * MyObject.hd, var3 - 22 * MyObject.hd, 0, var1);
                  }
               }
            }
         }
      }

   }
}

File: client/unity/Assets/Scripts/Assembly-CSharp/CellFarm.cs
public class CellFarm : SubObject
{
	public int xCell;

	public int yCell;

	public int idTree;

	public short time;

	public long tempTime;

	public bool isArid;

	public bool isWorm;

	public bool isGrass;

	public bool isSelected;

	public sbyte hervestPer;

	public sbyte vitalityPer;

	public sbyte status;

	public sbyte level;

	public int statusTree;

	public override void paint(MyGraphics g)
	{
		if ((float)(x * MyObject.hd) < AvCamera.gI().xCam - 10f || (float)(xCell * MyObject.hd) > AvCamera.gI().xCam + (float)Canvas.w + 10f)
		{
			return;
		}
		if (isSelected)
		{
			if (MyObject.hd == 2)
			{
				g.drawImage(FarmScr.imgFocusCel, (x - 13) * MyObject.hd, (y - 18) * MyObject.hd, 0);
			}
			else
			{
				g.drawImage(FarmScr.imgFocusCel, x - 11, y - 15, 0);
			}
		}
		int num = FarmScr.focusCell.x;
		int num2 = FarmScr.focusCell.y;
		if (xCell == num && yCell == num2)
		{
			Canvas.smallFontYellow.drawString(g, "lv" + level, x * MyObject.hd, (y - 44) * MyObject.hd, 2);
		}
		if (idTree == -1)
		{
			return;
		}
		TreeInfo treeByID = FarmData.getTreeByID(idTree);
		treeByID.paint(g, statusTree, x * MyObject.hd, y * MyObject.hd, MyGraphics.BOTTOM | MyGraphics.HCENTER);
		int num3 = treeByID.harvestTime * 60 + treeByID.dieTime * 60;
		if ((time > num3 && treeByID.dieTime > 0) || hervestPer == 100 || time < 0)
		{
			return;
		}
		if (isGrass)
		{
			g.drawImage(FarmScr.imgWorm_G[1], (x + 5) * MyObject.hd, (y - 2) * MyObject.hd, 3);
		}
		if (isWorm)
		{
			g.drawImage(FarmScr.imgWorm_G[0], (x - 7) * MyObject.hd, y * MyObject.hd, 3);
		}
		if (xCell == num && yCell == num2)
		{
			treeByID.paint(g, 7, num * 24 * MyObject.hd - 3, (num2 * 24 - 40) * MyObject.hd, MyGraphics.BOTTOM | MyGraphics.HCENTER);
			g.setColor(1);
			g.fillRect((num * 24 - 4) * MyObject.hd, (num2 * 24 - 38) * MyObject.hd, 31 * MyObject.hd, 4 * MyObject.hd);
			g.setColor(65280);
			g.fillRect((num * 24 - 3) * MyObject.hd, (num2 * 24 - 37) * MyObject.hd, vitalityPer * 30 / 100 * MyObject.hd, 3 * MyObject.hd);
			g.setColor(2512938);
			g.drawRect((num * 24 - 4) * MyObject.hd, (num2 * 24 - 38) * MyObject.hd, 31 * MyObject.hd, 4 * MyObject.hd);
			long num4 = treeByID.harvestTime * 60 * 60 - tempTime;
			long num5 = treeByID.harvestTime * 60 - time;
			string empty = string.Empty;
			if (num4 < 0)
			{
				num4 = 0L;
			}
			long num6 = num4 / 60 / 60;
			long num7 = num4 / 60 % 60;
			string text = empty;
			empty = text + num6 + ":" + num7;
			if (num5 <= 0)
			{
				empty = T.canHarvest;
			}
			Canvas.smallFontYellow.drawString(g, empty, (num * 24 + 5) * MyObject.hd, (num2 * 24 - 40) * MyObject.hd - AvMain.hSmall, 0);
			int num8 = time * 100 / (treeByID.harvestTime * 60);
			num8 = num8 * 30 / 100;
			if (num8 == 0)
			{
				num8 = 1;
			}
			if (num8 >= 30)
			{
				num8 = 29;
			}
			if (treeByID.harvestTime * 60 - time < 0)
			{
				num8 = 30;
			}
			g.setColor(1);
			g.fillRect((num * 24 - 4) * MyObject.hd, (num2 * 24 - 32) * MyObject.hd, 31 * MyObject.hd, 4 * MyObject.hd);
			g.setColor(16776960);
			g.fillRect((num * 24 - 3) * MyObject.hd, (num2 * 24 - 31) * MyObject.hd, num8 * MyObject.hd, 3 * MyObject.hd);
			g.setColor(2512938);
			g.drawRect((num * 24 - 4) * MyObject.hd, (num2 * 24 - 32) * MyObject.hd, 31 * MyObject.hd, 4 * MyObject.hd);
			int num9 = 0;
			if (isGrass)
			{
				num9 = 1;
				FarmScr.imgWormAndGrass.drawFrame(1, (num * 24 + 5 + (isWorm ? 6 : 0)) * MyObject.hd, (num2 * 24 - 22) * MyObject.hd, 0, g);
			}
			if (isWorm)
			{
				FarmScr.imgWormAndGrass.drawFrame(0, (num * 24 + 4 - num9 * 6) * MyObject.hd, (num2 * 24 - 22) * MyObject.hd, 0, g);
			}
		}
	}
}
